embedded/tbtree/tbtree.go

```diff
@@ -478,26 +480,54 @@ func (t *TBtree) cachePut(n node) {
 	t.nmutex.Lock()
 	defer t.nmutex.Unlock()
 
-	t.cache.Put(n.offset(), n)
+	r, _, _ := t.cache.Put(n.offset(), n)
+	if r != nil {
+		metricsCacheEvict.WithLabelValues(t.path).Inc()
+	}
 }
 
+var metricsCacheSizeStats = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "immudb_btree_cache_size",
+}, []string{"id"})
+
+var metricsCacheHit = promauto.NewCounterVec(prometheus.CounterOpts{
+	Name: "immudb_btree_cache_hit",
+}, []string{"id"})
+
+var metricsCacheMiss = promauto.NewCounterVec(prometheus.CounterOpts{
+	Name: "immudb_btree_cache_miss",
+}, []string{"id"})
+
+var metricsCacheEvict = promauto.NewCounterVec(prometheus.CounterOpts{
+	Name: "immudb_btree_cache_evict",
+}, []string{"id"})
+
 func (t *TBtree) nodeAt(offset int64) (node, error) {
 	t.nmutex.Lock()
 	defer t.nmutex.Unlock()
 
+	size := t.cache.EntriesCount()
+	metricsCacheSizeStats.WithLabelValues(t.path).Set(float64(size))
```

Comment:
is it needed every time a node is read?

Reply:
nope, I think we can update it only when modifying the cache - but in general it should be very lightweight operation, prometheus internals are highly optimized for such frequently changing metrics.

I've added some comments about that to https://github.com/codenotary/immudb/issues/1081